### PR TITLE
feat: implement resource metadata labels

### PR DESCRIPTION
**Pull request by smira**
Labels are simple key-value pairs attached to the resource metadata.
Metadata value can be still copied, labels "fork" themselves on update.

List request include new ListOption to specify label query: query terms
are combined with AND operator, and each term might check for label
existence or value match.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

**Review by Unix4ever — APPROVED** [review]
:cool:

File: pkg/resource/labels.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+// Raw returns the labels map.
+//
+// Label map should not be modified outside of the call.
+func (labels *Labels) Raw() map[string]string {
+	return labels.m
+}
+
+// Equal checks label for equality.
+func (labels Labels) Equal(other Labels) bool {
+	if labels.m == nil && other.m == nil {
```
Comment (DmitriyMV) [inline pkg/resource/labels.go:61]: Do we differentiate between nil map and empty map?

Comment (smira) [reply pkg/resource/labels.go:61]: that's a good question; also there's no way to delete labels right now, so there's no empty map ever.

Comment (smira) [reply pkg/resource/labels.go:61]: but I think we should handle that

Comment (smira) [reply pkg/resource/labels.go:78]: added the handling and a delete method

**Pull request merged**